rabbitmq/receive.js

```diff
@@ -71,11 +41,50 @@ wss.on("connection", ws => {
       if (deliverables.length > 0) {
         ws.send(JSON.stringify(deliverables.pop()));
       } else {
-        // webSockets[wsIndex].shouldDeliverDirectly = true;
         ws.send("empty");
       }
     } else if (message === "ready") {
       webSockets[wsIndex].shouldDeliverDirectly = true;
     }
   });
 });
+
+(async () => {
+  try {
+    const conn = await open;
+    const channel = await conn.createChannel();
+    const ok = await channel.assertQueue(queue, {
+      durable: false
+    });
+
+    if (ok) {
+      console.log(
+        " [*] RabbitMq initialized. Waiting for messages in %s. To exit press CTRL+C",
+        queue
+      );
+      channel.consume(
+        queue,
+        msg => {
```

Comment:
I will show you the way we do that in a function programming style.

```
// you add this to a file called fn.js
Function.prototype['∘'] = function (f) {
  return x => this(f(x));
};


const handleMsg = msg => {
  <same code as above>

  return deliverable;
}

const pushToWs = deliverable => {
  webSockets.forEach(({ws, shouldDeliverDirectly}, index) => {
    if (shouldDeliverDirectly) {
      ws.send(JSON.stringify(deliverable));
      webSockets[index].shouldDeliverDirectly = false;
    } else {
      deliverables.push(deliverable);
    }
  });
}

channel.consume(
  queue, 
  (pushToWs) ['∘'] (handleMsg),
  {noAck: true}
)
```

This is called functional composition. Essentially `(pushToWs) ['∘'] (handleMsg)` suggest that you handle the msg first and then the result (i.e. deliverable) is passed down to the `pushToWs` function.